### PR TITLE
update release notes script to automatically grab labels from the PR

### PR DESCRIPTION
**Pull request by bdhirsh**
Now that developers are asked to add release note labels to their PR's, we can update the release notes scripts to look at those labels when it categorizes commits.

Tested by running `python commitlist.py --create_new be7177751eda6a86720d2e3d72c6660646e81ada  1659a267f992156f1b636274cb4a3f94455e7c05`, and seeing `dataloader,bug fixes` in the output (which came from a labeled [dataloader PR](https://github.com/pytorch/pytorch/pull/74958))

Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #75020
* #72154

### CODE REVIEW

**Review by albanD — APPROVED** [review]
👍

File: scripts/release_notes/commitlist.py [hunk 1]
```diff
@@ -84,15 +84,29 @@ def keywordInFile(file, keywords):
...
         category = 'Uncategorized'
         topic = 'Untopiced'
 
+        # We ask contributors to label their PR's appropriately
+        # when they're first landed.
+        # Check if the labels are there first.
+        already_categorized = already_topiced = False
+        for label in labels:
+            if label.startswith('release notes: '):
+                category = label.split('release notes: ', 1)[1]
```
Comment (anjali411) [inline scripts/release_notes/commitlist.py:97]: are the categories used by the script in line with the category labels?

File: scripts/release_notes/commitlist.py [hunk 2]
```diff
@@ -84,15 +84,29 @@ def keywordInFile(file, keywords):
...
+
         # update this to check if each file starts with caffe2
         if 'caffe2' in title:
             return Commit(commit_hash, 'caffe2', topic, title)
         if '[codemod]' in title.lower():
             return Commit(commit_hash, 'skip', topic, title)
-        labels = features['labels']
         if 'Reverted' in labels:
             return Commit(commit_hash, 'skip', topic, title)
         if 'bc_breaking' in labels:
```
Comment (anjali411) [inline scripts/release_notes/commitlist.py:112]: `module: deprecation` is not a topic, so we should change it to 'deprecation'

**Review by anjali411 — APPROVED** [review]

**Comment by anjali411** [comment]
@bdhirsh you probably want to land the PR that's below this in the stack right?

**Comment by bdhirsh** [comment]
yep - I might have the wrong incantation, but I thought "@pytorchbot merge this please" is supposed to land the whole stack if you run it on the top PR
